installer.sh

```diff
@@ -59,33 +59,78 @@ confirm_force_install () {
     esac
 }
 
-
-# Which package management system are we using?
-if [[ -n "$(command -v dnf)" ]]; then
-    PACKAGE_MGR=$(command -v dnf)
-    PYTHON_PREIN="python3 python3-devel python3-setuptools git wget patch"
-    PYTHON_DEPS="python3-pip gcc gcc-c++ openssl-devel swig python3-pyyaml python3-m2crypto  python3-cryptography python3-tornado python3-simplejson python3-requests yaml-cpp-devel procps-ng"
-    PYTHON_PIPS="tornado pyzmq"
-    BUILD_TOOLS="openssl-devel libtool make automake pkg-config m4 libgcrypt-devel autoconf autoconf-archive libcurl-devel libstdc++-devel uriparser-devel dbus-devel gnulib-devel doxygen"
-elif [[ -n "$(command -v yum)" ]]; then
-    PACKAGE_MGR=$(command -v yum)
-    $PACKAGE_MGR -y install epel-release
-    PYTHON_PREIN="python36 python36-devel python36-setuptools python36-pip git wget patch openssl"
-    PYTHON_DEPS="gcc gcc-c++ openssl-devel swig python36-PyYAML python36-tornado python3-cryptography python36-simplejson python36-requests yaml-cpp-devel"
-    PYTHON_PIPS="tornado pyzmq m2crypto"
-    BUILD_TOOLS="openssl-devel libtool make automake m4 libgcrypt-devel autoconf autoconf-archive libcurl-devel libstdc++-devel uriparser-devel dbus-devel gnulib-devel doxygen"
-elif [[ -n "$(command -v apt-get)" ]]; then
-    PACKAGE_MGR=$(command -v apt-get)
-    PYTHON_PREIN="git patch"
-    PYTHON_DEPS="python3 python3-pip python3-dev python3-setuptools python3-zmq python3-tornado python3-cryptography python3-simplejson python3-requests gcc g++ libssl-dev swig python3-yaml wget"
-    PYTHON_PIPS="m2crypto"
-    BUILD_TOOLS="build-essential libtool automake pkg-config m4 libgcrypt20-dev uthash-dev autoconf autoconf-archive libcurl4-gnutls-dev gnulib doxygen libdbus-1-dev"
-    $PACKAGE_MGR update
-else
-   echo "No recognized package manager found on this system!" 1>&2
-   exit 1
+# Determine distibution (using systemd standard `os-release`):
+if [ -f /etc/os-release ]; then
+        . /etc/os-release
+    else
+        echo "Not able to determine your OS or Distribution"
+        exit 1
 fi
 
+NEED_BUILD_TOOLS=0
```

Comment:
We should define defaults for anything set below that isn't always set to a value.  For instance: 

````bash 
CENTOS7_TSS_FLAGS=
GOPKG=
HAS_GO_PKG=0
NEED_BUILD_TOOLS=0
NEED_PYTHON_DIR=
PYTHON_PIPS=
TPM2_TOOLS_PKGS=
````

Reply:
great suggestion!